mm2src/coins/eth.rs

```diff
@@ -2542,14 +2840,217 @@ impl EthCoin {
                             0.into(),
                             Action::Call(swap_contract_address),
                             data,
-                            U256::from(150_000),
+                            U256::from(ETH_GAS),
                         ))
                     }
                 }))
             },
         }
     }
 
+    fn watcher_spend_hash_time_locked_payment(
+        &self,
+        payment: SignedEthTx,
+        _secret_hash: &[u8],
+        secret: &[u8],
+        taker_pub: &[u8],
+    ) -> EthTxFut {
+        let spend_func = try_tx_fus!(SWAP_CONTRACT.function("watcherSpend"));
+        let clone = self.clone();
+        let secret_vec = secret.to_vec();
+        let taker_addr = addr_from_raw_pubkey(taker_pub).unwrap();
+        let swap_contract_address = match payment.action.clone() {
```

Comment:
You can avoid using `clone` here because `Address` implements `Copy`:
```rust
        let swap_contract_address = match &payment.action {
            Call(address) => *address,
            Create => {
                return Box::new(futures01::future::err(TransactionErr::Plain(ERRL!(
                    "Invalid payment action: the payment action cannot be create"
                ))))
            },
        };
```